- Fix build with OpenEXR>=3
- Switch to C++17
- Update Cmake and deps, Fix Freetype and FontConfig Linkage,
  from upstream


Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -105,11 +105,12 @@ endif()
 message(STATUS "Release build: ${RELEASE_BUILD}")
 
 # use CPP-11
-if (CMAKE_VERSION VERSION_LESS "3.1")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-else ()
-    set (CMAKE_CXX_STANDARD 11)
-endif ()
+#if (CMAKE_VERSION VERSION_LESS "3.1")
+#    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+#else ()
+#    set (CMAKE_CXX_STANDARD 11)
+#endif ()
+set (CMAKE_CXX_STANDARD 17)
 
 ############
 #############
@@ -391,8 +392,14 @@ endif ()
 ## Check for OpenEXR
 ##
 
-macro_optional_find_package(OpenEXR)
-macro_bool_to_01(OPENEXR_FOUND HAVE_OPENEXR)
+macro_optional_find_package(Imath 3.0 CONFIG QUIET)
+if(TARGET Imath::Imath)
+    set(OPENEXR_LIBRARIES Imath::Imath)
+    set(OpenEXR_FOUND TRUE)
+else()
+    macro_optional_find_package(OpenEXR)
+endif()
+macro_bool_to_01(OpenEXR_FOUND HAVE_OPENEXR)
 
 ##
 ## Test for GNU Scientific Library
@@ -641,8 +648,8 @@ if(NOT WIN32 AND NOT APPLE)
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
@@ -912,6 +919,8 @@ set(KOPAGEAPP_INCLUDES ${TEXTLAYOUT_INCLUDES}
     ${CMAKE_SOURCE_DIR}/libs/pageapp
     ${CMAKE_SOURCE_DIR}/libs/pageapp/commands
     ${CMAKE_BINARY_DIR}/libs/pageapp )
+
+
 
 #############################################
 ####           filter libraries          ####
